spec/unit/provider/package/yum_spec.rb

```diff
@@ -2279,8 +2279,35 @@ def reset_instance
       expect(@provider).to receive(:yum_command).with(
         "-d0 -e0 -y install cups-1.2.4-11.19.el5.x86_64 vim-1.0"
       )
-      @provider.install_package(%w{cups vim}, ["1.2.4-11.19.el5", "1.0"])
+      @provider.install_package(%w{cups.x86_64 vim}, ["1.2.4-11.19.el5", "1.0"])
     end
 
+    it "should run yum install with the package name, version, and arch when names have arch" do
+      @new_resource = Chef::Resource::Package.new(["cups.x86_64", "cups.i386"])
```

Comment:
If you replace this with:

```ruby
      @new_resource = Chef::Resource::YumPackage.new(["cups", "vim"])
      @new_resource.arch [ "x86_64", nil ]
```

it needs to work identically, but its blowing up on me.  IDK if its something stubbed that isn't being stubbed properly though